datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/policy/GetGrantedPrivilegesResolver.java

```diff
@@ -44,9 +50,32 @@ public CompletableFuture<Privileges> get(final DataFetchingEnvironment environme
     if (context.getAuthorizer() instanceof AuthorizerChain) {
       DataHubAuthorizer dataHubAuthorizer =
           ((AuthorizerChain) context.getAuthorizer()).getDefaultAuthorizer();
-      List<String> privileges = dataHubAuthorizer.getGrantedPrivileges(actor, resourceSpec);
+
       return GraphQLConcurrencyUtils.supplyAsync(
-          () -> Privileges.builder().setPrivileges(privileges).build(),
+          () -> {
+            try {
+              PolicyEngine.PolicyGrantedPrivileges evalResult =
+                  dataHubAuthorizer.getGrantedPrivileges(actor, resourceSpec);
+
+              List<PolicyEvaluationDetail> evaluationDetailList = null;
+
+              if (input.getIncludeEvaluationDetails()) {
```

Comment:
Curiosity question, mainly to @david-leifker & @RyanHolstien 

Shouldn't GraphQL be able to determine whether to compute the evaluation details only if the caller requests that information in the return?

Reply:
Yeah generally the way that errors work in GraphQL is that you can throw errors per resolver and bundle them up at the top layer if they're not hard failures. We just throw hard failures. Changing this would be a significant refactor.